nav2_system_tests/src/system/nav_through_poses_tester_error_msg_node.py

```diff
@@ -0,0 +1,333 @@
+        future = mgr_client.call_async(req)
+        try:
+            self.info_msg('Shutting down navigation lifecycle manager...')
+            rclpy.spin_until_future_complete(self, future)
+            future.result()
+            self.info_msg('Shutting down navigation lifecycle manager complete.')
+        except Exception as e:  # noqa: B902
+            self.error_msg(f'Service call failed {e!r}')
+        transition_service = 'lifecycle_manager_localization/manage_nodes'
+        mgr_client = self.create_client(ManageLifecycleNodes, transition_service)
+        while not mgr_client.wait_for_service(timeout_sec=1.0):
+            self.info_msg(f'{transition_service} service not available, waiting...')
+
+        req = ManageLifecycleNodes.Request()
+        req.command = ManageLifecycleNodes.Request().SHUTDOWN
+        future = mgr_client.call_async(req)
+        try:
+            self.info_msg('Shutting down localization lifecycle manager...')
+            rclpy.spin_until_future_complete(self, future)
+            future.result()
+            self.info_msg('Shutting down localization lifecycle manager complete')
+        except Exception as e:  # noqa: B902
+            self.error_msg(f'Service call failed {e!r}')
+
+    def wait_for_initial_pose(self):
+        self.initial_pose_received = False
+        while not self.initial_pose_received:
+            self.info_msg('Setting initial pose')
+            self.setInitialPose()
+            self.info_msg('Waiting for amcl_pose to be received')
+            rclpy.spin_once(self, timeout_sec=1)
+
+
+def run_all_tests(robot_tester):
+    pose_out_of_bounds = Pose()
+    pose_out_of_bounds.position.x = 2000.0
+    pose_out_of_bounds.position.y = 4000.0
+    pose_out_of_bounds.position.z = 0.0
+    pose_out_of_bounds.orientation.x = 0.0
+    pose_out_of_bounds.orientation.y = 0.0
+    pose_out_of_bounds.orientation.z = 0.0
+    pose_out_of_bounds.orientation.w = 1.0
+
+    result = True
+    if result:
+        robot_tester.wait_for_node_active('amcl')
+        robot_tester.wait_for_initial_pose()
+        robot_tester.wait_for_node_active('bt_navigator')
+
+        result = robot_tester.runNavigateAction(
+            goal_pose=pose_out_of_bounds,
+            behavior_tree="",
+            expected_error_code=304,
+            expected_error_msg=('GridBasedplugin failed to plan from '
+                                '(-2.00, -0.50) to (2000.00, 4000.00): '
+                                '"Goal Coordinates of(2000.000000, 4000.000000) '
+                                'was outside bounds"'))
+    if result:
+        result = robot_tester.runNavigateAction(
+            goal_pose=pose_out_of_bounds,
+            behavior_tree="behavior_tree_that_does_not_exist.xml",
+            expected_error_code=901,
+            expected_error_msg=('Error loading XML file: behavior_tree_that_does_not_exist.xml. '
+                                'Navigation canceled.'))
+
```

Comment:
@SteveMacenski here are some error message related system tests (see above this comment).  Is this on the right track of what you are after?  Got any ideas of what else you want checked?